Mapping/Proxy/ProxyLoader.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Mapping\Proxy;
+
+use Symfony\Component\Config\ConfigCache;
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Loads proxy documents.
+ */
+class ProxyLoader
+{
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @var Filesystem
+     */
+    private $fileSystem;
+
+    /**
+     * @var bool
+     */
+    private $debug;
+
+    /**
+     * Constructor.
+     *
+     * @param string $cacheDir To load proxies to.
+     * @param bool   $debug    Whether debugging is enabled or not.
+     */
+    public function __construct($cacheDir, $debug = false)
+    {
+        $this->cacheDir = $cacheDir;
+        $this->debug = $debug;
+    }
+
+    /**
+     * Loads document proxy class into cache.
+     *
+     * @param \ReflectionClass $reflectionClass
+     *
+     * @return string Proxy document path.
+     */
+    public function load(\ReflectionClass $reflectionClass)
+    {
+        $cacheBundleDir = $this->getCacheDir($reflectionClass->getName());
+
+        if (!$this->getFilesystem()->exists($cacheBundleDir)) {
+            $this->getFilesystem()->mkdir($cacheBundleDir, 0700);
```

Comment:
everything is fine here, directories should have x flag set, but there are some rare cases where permissions should be more strict